### PR TITLE
Downloader: raise ChunkStreamingError if partial download detected

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This PR fix the behavior of downloader that reports incompleted download as HashVerificationError, now in-completed download will be reported as `ChunkStreamingError`.

## Changes
Now `Downloader.download` take an optional `size` param, the downloader will check the downloaded file's size against the input size, and detect whether there is an in-completed download.

## Ticket
https://tier4.atlassian.net/browse/T4PB-20330

### CODE REVIEW

**Comment by obi-t4** [comment]
@Bodong-Yang Can you add unit test?

**Comment by Bodong-Yang** [comment]
> @Bodong-Yang Can you add unit test?

Done in [f90e163](https://github.com/tier4/ota-client/pull/157/commits/f90e163ae60b658c5b300a840fc2e99d1bd469b5).

File: tests/test_downloader.py [hunk 1]
```diff
@@ -101,6 +103,25 @@ def test_download_server_error(
...
+
+        _downloader = Downloader()
+        _target_path = tmp_path / self.TEST_FILE
+
+        with pytest.raises(ChunkStreamingError) as e:
+            _downloader.download(
+                self.TEST_FILE,
+                _target_path,
+                digest=self.TEST_FILE_SHA256,
+                size=random.randint(
```
Comment (obi-t4) [inline tests/test_downloader.py:117]: I'd like to remove randomness from the test.

Comment (obi-t4) [reply tests/test_downloader.py:117]: `size=self.TEST_FILE_SIZE // 2` is enough.

Comment (Bodong-Yang) [reply tests/test_downloader.py:117]: OK!

Comment (Bodong-Yang) [reply tests/test_downloader.py]: Updated in [13ce31b](https://github.com/tier4/ota-client/pull/157/commits/13ce31bf903a3731d7b66fd32783579028517ae9).

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**